java/vortex-spark/src/main/java/dev/vortex/spark/write/VortexDataWriter.java

```diff
@@ -218,9 +222,19 @@ private void populateVector(FieldVector vector, DataType dataType, InternalRow r
                         .toJavaBigDecimal();
                 ((DecimalVector) vector).set(rowIndex, decimal);
             }
+        } else if (dataType instanceof ArrayType) {
+            ArrayType arrayType = (ArrayType) dataType;
+            ArrayData data = row.getArray(fieldIndex);
+            ListVector listVector = ((ListVector) vector);
+            int writtenElements = listVector.getElementEndIndex(listVector.getLastSet());
+            listVector.startNewValue(rowIndex);
+            for (int i = 0; i < data.numElements(); i++) {
+                populateVector(listVector.getDataVector(), arrayType.elementType(), data, i, writtenElements + i);
```

Comment:
you need to be a tad cute about how you populate the inner elements vector that the ListVector wraps